internal/serve/httphandler/user_handler.go

```diff
@@ -224,6 +223,18 @@ func (h UserHandler) CreateUser(rw http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	err = services.SendInvitationMessage(ctx, h.MessengerClient, h.Models,
+		services.SendInvitationMessageOptions{
+			FirstName: newUser.FirstName,
+			Email:     newUser.Email,
+			Role:      newUser.Roles[0],
+			UIBaseURL: *tnt.SDPUIBaseURL,
+		})
+	if err != nil {
+		httperror.InternalError(ctx, "Cannot send invitation message", err, nil).Render(rw)
+		return
+	}
```

Comment:
Hmmmm, let's discuss the response code used here? I don't think a 500 would be the best option because the tenant was created, along with its owner. I believe we should log the error and send a special response for this use case.

I did some research and some people use the 207 status code, which means multi-status. A special json would be needed to convey what happened, like:
```json
{
    "status": "multi-status",
    "results": [
        {
            "operation": "provision_tenant",
            "status": "success",
            "status_code": 201,
            "tenant": {...}
        },
        {
            "operation": "send_invitation_email",
            "status": "failed",
            "error": "{the error string}"
        }
    ]
}
```

207 is not widely used, so I'm open to discuss this further.

Reply:
@marwen-abid: would you like to chime in?

Reply:
I didn't know about the 207 response but I doubt users of the API would want to handle this esoteric code.

The API is used by the Host admin, so it should be the same role/organization that also monitors the SDP logs. It makes sense to return 201 to say that the tenant was created successfully in this case, then log the error and submit it to the crashtracker. The host's monitoring should catch the fact that the invitation isn't sent properly and address that.

Reply:
Agreed, 201 with a log + crash reporting should be easier to handle than a 207.

Reply:
Reviewed that this was addressed